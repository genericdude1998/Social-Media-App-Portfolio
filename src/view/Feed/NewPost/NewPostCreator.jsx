import React from 'react';
import { connect } from 'react-redux';
import { mapNPCDispatchToProps, mapNPCStateToProps } from '../../mappers/mappers';

const NewPostCreator = ({onSendNewPost, newPostError}) => {
    return (
        <div>
            <button onClick={onSendNewPost()}>Create Post</button>
        </div>
    );
}

const ConnectedNPC = connect(mapNPCStateToProps, mapNPCDispatchToProps)(NewPostCreator);

export default ConnectedNPC;
