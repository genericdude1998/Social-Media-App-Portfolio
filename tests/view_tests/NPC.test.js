import React from 'react';
import { shallow } from 'enzyme';
import NewPostCreator from '../../src/view/Feed/NewPost/NewPostCreator';
import {mockContent, mockErrorMessage} from '../mockValues';

jest.mock('react-router', () => {
    const originalModule = jest.requireActual('react-router');
  
    //Mock the default export and named export 'foo'
    return {
        __esModule: true,
        ...originalModule,
        useNavigate: () => jest.fn(),
    };
});

let wrapper = shallow(
    <NewPostCreator.WrappedComponent.WrappedComponent 
        onSendNewPost={jest.fn()}
        onSetPostContent={jest.fn()}
        newPostError={mockErrorMessage}
        content={mockContent}
        theme={'light'}
    />
);

describe('NCC', () => {
    it('should matchSnapshot when light theme', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should matchSnapshot when dark theme', () => {
        wrapper.setProps({theme: 'dark'})
        expect(wrapper).toMatchSnapshot();
    });
});