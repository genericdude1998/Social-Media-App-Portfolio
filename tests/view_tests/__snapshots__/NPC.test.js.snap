// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NCC should matchSnapshot when dark theme 1`] = `
<div
  className="baseNPC dark"
>
  <h1
    className="baseNPCHeader"
  >
    Type the content of your comment
  </h1>
  <form
    className="baseNPCForm dark"
  >
    <label
      className="baseNPCLabel dark"
    >
      Content:
    </label>
    <textarea
      className="baseNPCInput dark"
      onChange={[MockFunction]}
    />
    <button
      className="baseNPCButton dark"
      type="submit"
    >
      Create Post!
    </button>
  </form>
</div>
`;

exports[`NCC should matchSnapshot when light theme 1`] = `
<div
  className="baseNPC null"
>
  <h1
    className="baseNPCHeader"
  >
    Type the content of your comment
  </h1>
  <form
    className="baseNPCForm null"
  >
    <label
      className="baseNPCLabel null"
    >
      Content:
    </label>
    <textarea
      className="baseNPCInput null"
      onChange={[MockFunction]}
    />
    <button
      className="baseNPCButton null"
      type="submit"
    >
      Create Post!
    </button>
  </form>
</div>
`;
